app/plugin.go

```diff
@@ -756,8 +758,11 @@ func (a *App) processPrepackagedPlugin(pluginPath *pluginSignaturePath) (*plugin
 		return nil, errors.Wrapf(err, "Failed to get prepackaged plugin %s", pluginPath.path)
 	}
 
-	// Skip installing the plugin at all if automatic prepackaged plugins is disabled.
-	if !*a.Config().PluginSettings.AutomaticPrepackagedPlugins {
+	pluginsEnvironment := a.GetPluginsEnvironment()
+
+	// Skip installing the plugin at all if automatic prepackaged plugins is disabled
+	// or if the plugin is has not been enabled.
+	if !(*a.Config().PluginSettings.AutomaticPrepackagedPlugins && pluginsEnvironment.IsActive(plugin.Manifest.Id)) {
```

Comment:
Two comments:
* I'm not sure `IsActive` is the right bit to check here. I think that will be true only if the plugin is installed and running. I think we need to check the configuration here, and determine if the plugin has been enabled or not.
* I personally find `!(condition && condition)` hard to parse: can we break these out explicitly instead?

Reply:
I've updated to use `PluginSettings.PluginStates` from the config.